src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3304,6 +3298,7 @@ private static bool IsNegativeConstantForArraySize(BoundExpression expression)
                 if (initializerArgumentListOpt != null)
                 {
                     this.BindArgumentsAndNames(initializerArgumentListOpt, diagnostics, analyzedArguments, allowArglist: true);
+                    diagnostics = analyzedArguments.AvoidCascading(diagnostics);
```

Comment:
If you use the same name twice (`M(x: 1, x: 2)`), then error from `BindArgumentsAndNames` is all you need. The error(s) from overload resolution will be noise.

I re-ran the tests without this change, to get some examples:

```C#
  static void M(int x, int y, int z)
    {
    }
    static void Main()
    {
        M(x: 1, x: 2, 3);
    }
```
produces:
```
 // (9,17): error CS1740: Named argument 'x' cannot be specified multiple times
                //         M(x: 1, x: 2, 3);
                Diagnostic(ErrorCode.ERR_DuplicateNamedArgument, "x").WithArguments("x").WithLocation(9, 17),
                // (9,23): error CS1738: Named argument specifications must appear after all fixed arguments have been specified. Please use language version 7.2 or greater to allow non-trailing named arguments.
                //         M(x: 1, x: 2, 3);
                Diagnostic(ErrorCode.ERR_NamedArgumentSpecificationBeforeFixedArgument, "3").WithArguments("7.2").WithLocation(9, 23),
                // (9,17): error CS8321: Named argument 'x' is used out-of-position but is followed by an unnamed argument
                //         M(x: 1, x: 2, 3);
                Diagnostic(ErrorCode.ERR_BadNonTrailingNamedArgument, "x").WithArguments("x").WithLocation(9, 17) // this cascading diagnostic could be avoided
```

```C#
 static void M()
    {
        M("""",
            arg: 0,
            arg: null);
    }
```
produces:
```
  // (7,13): error CS1740: Named argument 'arg' cannot be specified multiple times
                //             arg: null);
                Diagnostic(ErrorCode.ERR_DuplicateNamedArgument, "arg").WithArguments("arg").WithLocation(7, 13),
                // (5,9): error CS1501: No overload for method 'M' takes 3 arguments
                //         M("",
                Diagnostic(ErrorCode.ERR_BadArgCount, "M").WithArguments("M", "3").WithLocation(5, 9) // this cascading diagnostic could be avoided
```

The cascading diagnostics are not awful. I could be convinced (if so, I'd open a follow-up issue since Aleksey had requested the change).